Addon classifiers not used when locating SDK feature model

When using addons the SDK feature model cannot be found because its configured classifier is not considered, which gives error messages like this:

```
[ERROR] Failed to execute goal com.adobe.aem:aemanalyser-maven-plugin:1.0.1-SNAPSHOT:project-analyse (aem-analyser) on project testing-forms-cloud.all: Execution aem-analyser of goal com.adobe.aem:aemanalyser-maven-plugin:1.0.1-SNAPSHOT:project-analyse failed: Unable to read feature file /Users/david/.m2/repository/com/adobe/aem/aem-forms-sdk-api/2021.05.12.04/aem-forms-sdk-api-2021.05.12.04.jar for com.adobe.aem:aem-forms-sdk-api:2021.05.12.04: javax.json.stream.JsonParsingException: Unexpected character 'P' (Codepoint: 80) on [lineNumber=1, columnNumber=2, streamOffset=1]. Reason is [[Expected structural character or digit or 't' or 'n' or 'f' or '-']] -> [Help 1]
```

--- a/aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/AemAggregator.java
+++ b/aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/AemAggregator.java
@@ -60,9 +60,9 @@ public class AemAggregator {
     private ArtifactProvider artifactProvider;
 
     private FeatureProvider featureProvider;
-    
+
     private ArtifactId projectId;
-    
+
     private ArtifactId sdkId;
 
     private List<ArtifactId> addOnIds;
@@ -212,7 +212,7 @@ public class AemAggregator {
     throws IOException {
         // get run modes from converter output
         final Properties runmodes = getRunmodeMappings();
-        
+
         final Map<String, List<Feature>> aggregates = new HashMap<>();
 
         Map<String, Set<String>> toCreate = getUserAggregatesToCreate(runmodes);
@@ -228,7 +228,7 @@ public class AemAggregator {
 
     Map<String, Set<String>> getUserAggregatesToCreate(final Properties runmodes) throws IOException {
         try {
-            return AemAnalyserUtil.getAggregates(runmodes);            
+            return AemAnalyserUtil.getAggregates(runmodes);
         } catch ( final IllegalArgumentException iae) {
             throw new IOException(iae.getMessage());
         }
@@ -250,7 +250,7 @@ public class AemAggregator {
             list.add(sdkFeature);
             if ( this.getAddOnIds() != null ) {
                 for(final ArtifactId id : this.getAddOnIds()) {
-                    final Feature feature = getFeatureProvider().provide(id);
+                    final Feature feature = getFeatureProvider().provide(id.changeType(FEATUREMODEL_TYPE));
                     if ( feature == null ) {
                         throw new IOException("Unable to find addon feature for " + id.toMvnId());
                     }
@@ -261,7 +261,7 @@ public class AemAggregator {
 
         return aggregates;
     }
-    
+
     Map<String, List<Feature>> getFinalAggregates(final Map<String, List<Feature>> userAggregate,
             final Map<String, Feature> projectFeatures) throws IOException {
         final Map<String, List<Feature>> aggregates = new HashMap<>();
@@ -278,11 +278,11 @@ public class AemAggregator {
 
         return aggregates;
     }
-    
+
     private String getProductAggregateName(final boolean author) {
         return "product-aggregated-" + (author ? "author" : "publish");
     }
-    
+
 
     List<Feature> aggregate(final Map<String, List<Feature>> aggregates, final Mode mode,
         final Map<String, Feature> projectFeatures) throws IOException {
@@ -293,7 +293,7 @@ public class AemAggregator {
             logger.info("Building aggregate {}...", aggregate.getKey());
 
             final BuilderContext builderContext = new BuilderContext(new FeatureProvider(){
-           
+
                 @Override
                 public Feature provide(final ArtifactId id) {
                     // check in selection
@@ -330,8 +330,8 @@ public class AemAggregator {
             builderContext.addConfigsOverrides(Collections.singletonMap("*", "MERGE_LATEST"));
 
             final ArtifactId newFeatureID = this.getProjectId().changeClassifier(aggregate.getKey()).changeType(FEATUREMODEL_TYPE);
-            
-            final Feature feature = FeatureBuilder.assemble(newFeatureID, builderContext, 
+
+            final Feature feature = FeatureBuilder.assemble(newFeatureID, builderContext,
                   aggregate.getValue().toArray(new Feature[aggregate.getValue().size()]));
 
             try ( final Writer writer = new FileWriter(new File(this.getFeatureOutputDirectory(), aggregate.getKey().concat(".json")))) {

--- a/aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/VersionUtil.java
+++ b/aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/VersionUtil.java
@@ -49,7 +49,7 @@ public class VersionUtil {
 
     private final List<String> versionWarnings = new ArrayList<>();
 
-    public VersionUtil(final Log log, 
+    public VersionUtil(final Log log,
             final MavenProject project,
             final ArtifactHandlerManager artifactHandlerManager,
             final ArtifactMetadataSource artifactMetadataSource,
@@ -96,6 +96,8 @@ public class VersionUtil {
                     useVersion = foundVersion;
                 }
                 addonSDK = addonSDK.changeVersion(useVersion);
+                if (addon.classifier != null)
+                    addonSDK = addonSDK.changeClassifier(addon.classifier);
 
                 this.log.info("Using Add-On for analysis: " + addonSDK);
 
@@ -105,7 +107,7 @@ public class VersionUtil {
 
         return result;
     }
-    
+
     /**
      * Get the SDK artifact id
      * @return The artifact id of the SDK
@@ -126,7 +128,7 @@ public class VersionUtil {
                 if ( dependencySdk == null && configuredVersion == null ) {
                     throw new MojoExecutionException("Unable to find SDK artifact in dependencies or dependency management: "
                                         + Constants.SDK_GROUP_ID + ":" + configuredArtifactId);
-                }    
+                }
             }
         } else {
             // first search prerelease SDK
@@ -152,7 +154,7 @@ public class VersionUtil {
             dep.setGroupId(Constants.SDK_GROUP_ID);
             dep.setArtifactId(useArtifactId);
             dep.setVersion(dependencySdk == null ? "1.0" : dependencySdk.getVersion());
-            
+
             final String foundVersion = useDependencyVersions ? null : getLatestVersion(dep);
             if ( foundVersion == null && dependencySdk == null ) {
                 throw new MojoExecutionException("Unable to find SDK artifact in dependencies or dependency management: "
@@ -180,19 +182,19 @@ public class VersionUtil {
         final ArtifactVersion fv = new DefaultArtifactVersion(foundVersion);
         return fv.compareTo(ev) > 0;
     }
- 
+
 
     /**
      * Get the artifact id from the dependencies of the project
-     * @param groupId The group id 
+     * @param groupId The group id
      * @param artifactId The artifact id
      * @return The artifact id or {@code null}
      * @throws MojoExecutionException On error
      */
-    ArtifactId getArtifactIdFromDependencies(final String groupId, final String artifactId) 
+    ArtifactId getArtifactIdFromDependencies(final String groupId, final String artifactId)
     throws MojoExecutionException {
         final List<Dependency> allDependencies = new ArrayList<>();
-        
+
         if (project.getDependencies() != null) {
             allDependencies.addAll(project.getDependencies());
         }
@@ -212,7 +214,7 @@ public class VersionUtil {
 
     String getLatestVersion( final Dependency dependency )
         throws MojoExecutionException {
-        
+
         try {
             final Artifact artifact = new DefaultArtifact(dependency.getGroupId(),
                 dependency.getArtifactId(),
@@ -224,15 +226,15 @@ public class VersionUtil {
 
             final List<ArtifactVersion> versions =
                 artifactMetadataSource.retrieveAvailableVersions( artifact, localRepository, remoteArtifactRepositories );
-    
+
             ArtifactVersion latest = null;
             for ( final ArtifactVersion candidate : versions ) {
                 if ( ArtifactUtils.isSnapshot( candidate.toString() ) ) {
                     continue;
                 }
-        
+
                 if ( latest == null ) {
-                    latest = candidate;                
+                    latest = candidate;
                 } else {
                     if ( candidate.compareTo(latest) > 0 ) {
                         latest = candidate;

